pkg/trace/traceutil/processed_trace.go

```diff
@@ -18,3 +18,15 @@ type ProcessedTrace struct {
 	TracerHostname         string
 	ClientDroppedP0sWeight float64
 }
+
```

Comment:
```suggestion
// Clone creates a slightly deep (but not fully deep) copy of ProcessedTrace, cloning p,
// p.TraceChunk, and p.Root. This means it is safe to modify the returned ProcessedTrace's - call
// it pt - fields along with fields in pt.TraceChunk and fields in pt.Root.
// 
// The most important consequence of this is that the TraceChunk's Spans field can be assigned,
// *BUT* the Spans value itself should not be modified. i.e. This is ok:
//   pt2 := pt.Clone()
//   pt2.TraceChunk.Spans = make([]*pb.Span)
// but this is NOT ok:
//   pt2 := pt.Clone()
//   pt2.TraceChunk.Spans[0] = &pb.Span{} // This will be visible in pt.
```